websockets-jsr/src/main/java/io/undertow/websockets/jsr/ServerWebSocketContainer.java

```diff
@@ -356,14 +356,19 @@ public void handleDone(WebSocketChannel data, Object attachment) {
             synchronized (clientEndpoints) {
                 configured = clientEndpoints.get(endpointInstance.getClass());
                 if(configured == null) {
-                    clientEndpoints.put(endpointInstance.getClass(), configured = new ConfiguredClientEndpoint());
+                    // make sure to create an instance of AnnotatedEndpoint
+                    clientEndpoints.put(endpointInstance.getClass(),
+                            getClientEndpoint(endpointInstance.getClass(), false));
```

Comment:
Please update configured = or else the method call that uses it in the sequence will receive null

Reply:
Have updated the code so it uses the provided endpoint instance in case the initialization of configured fails. Please review.